### PR TITLE
Disabling istio sidecars injection

### PR DESCRIPTION
**Pull request by dariusztutaj**
**Description**

Changes proposed in this pull request:

- Default sidecar injection policy set to disabled
- Updated tutorials, troubleshooting guides, adjusted to new policy

**Related issue(s)**
#1126
#14911

### CODE REVIEW

File: docs/01-overview/main-areas/service-mesh/smsh-99-useful-links.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ If you're interested in learning more about Service Mesh in Kyma, follow these l
 
 - Perform some simple and more advanced tasks:
 
-  - [Disable Istio Sidecar Proxy Injection](../../../04-operation-guides/operations/smsh-01-istio-disable-sidecar-injection.md)
+  - [Disable Istio Sidecar Proxy Injection](../../../04-operation-guides/operations/smsh-01-istio-enable-sidecar-injection.md)
```
Comment (barchw) [inline docs/01-overview/main-areas/service-mesh/smsh-99-useful-links.md:13]:
```suggestion
  - [Enable Istio Sidecar Proxy Injection](../../../04-operation-guides/operations/smsh-01-istio-enable-sidecar-injection.md)
```

**Review by barchw — CHANGES_REQUESTED** [review]

File: docs/04-operation-guides/troubleshooting/service-mesh/smsh-03-istio-no-sidecar.md [hunk 2]
```diff
@@ -8,28 +8,28 @@ A Pod doesn't have a sidecar but you did not disable sidecar injection on purpos
 
 ## Cause
 
-By default, Kyma has sidecar injection enabled - a sidecar is injected to every Pod in a cluster without the need to add any labels. For more information, read the document about [disabling Istio sidecar proxy injection](../../operations/smsh-01-istio-disable-sidecar-injection.md).
+By default, Kyma has sidecar injection disabled - a sidecar is not injected to any Pod in a cluster. For more information, read the document about [enabling Istio sidecar proxy injection](../../operations/smsh-01-istio-enable-sidecar-injection.md).
```
Comment (SamehMKhattab) [inline docs/04-operation-guides/troubleshooting/service-mesh/smsh-03-istio-no-sidecar.md:11]:
```suggestion
By default, Kyma has sidecar injection disabled - there is no automatic sidecar injection into any Pod in a cluster. For more information, read the document about [enabling Istio sidecar proxy injection](../../operations/smsh-01-istio-enable-sidecar-injection.md).
```

File: docs/04-operation-guides/troubleshooting/service-mesh/smsh-04-istio-sidecar-version.md [hunk 3]
```diff
@@ -8,7 +8,7 @@ You upgraded Kyma and mesh connectivity is broken.
 
 ## Cause
 
-By default, Kyma has sidecar injection enabled - a sidecar is injected to every Pod in a cluster without the need to add any labels. For more information, read the document about [disabling Istio sidecar proxy injection](../../operations/smsh-01-istio-disable-sidecar-injection.md).
+By default, Kyma has sidecar injection disabled - a sidecar is not injected to any Pod in a cluster. For more information, read the document about [disabling Istio sidecar proxy injection](../../operations/smsh-01-istio-enable-sidecar-injection.md).
```
Comment (SamehMKhattab) [inline docs/04-operation-guides/troubleshooting/service-mesh/smsh-04-istio-sidecar-version.md:11]:
```suggestion
By default, Kyma has sidecar injection disabled - there is no automatic sidecar injection into any Pod in a cluster. For more information, read the document about [enabling Istio sidecar proxy injection](../../operations/smsh-01-istio-enable-sidecar-injection.md).
```

File: docs/05-technical-reference/00-architecture/svls-01-architecture.md [hunk 4]
```diff
@@ -6,7 +6,9 @@ Serverless relies heavily on Kubernetes resources. It uses [Deployments](https:/
 
 ![Serverless architecture](./assets/svls-architecture.svg)
 
->**CAUTION:** Serverless imposes some requirements on the setup of Namespaces. If you create a new Namespace, do not disable sidecar injection in it as Serverless requires Istio for other resources to communicate with Functions correctly. Also, if you apply custom [LimitRanges](https://kubernetes.io/docs/concepts/policy/limit-range/) for a new Namespace, they must be higher than or equal to the [limits for building Jobs' resources](../05-configuration-parameters/svls-01-serverless-chart.md).
+**CAUTION:** Serverless imposes some requirements on the setup of Namespaces. For example, if you create a new Namespace you have to enable sidecar injection in it, as Serverless requires Istio for other resources to communicate with Functions correctly. Also, if you apply custom [LimitRanges](https://kubernetes.io/docs/concepts/policy/limit-range/) for a new Namespace, they must be higher than or equal to the [limits for building Jobs' resources](../05-configuration-parameters/svls-01-serverless-chart.md).
```
Comment (SamehMKhattab) [inline docs/05-technical-reference/00-architecture/svls-01-architecture.md:9]:
```suggestion
**CAUTION:** Serverless imposes some requirements on the setup of Namespaces. For example, if you create a new Namespace you have to enable sidecar injection in it, as Serverless needs Istio for other resources to communicate with Functions correctly. Also, if you apply custom [LimitRanges](https://kubernetes.io/docs/concepts/policy/limit-range/) for a new Namespace, they must be higher than or equal to the [limits for building Jobs' resources](../05-configuration-parameters/svls-01-serverless-chart.md).
```

File: docs/05-technical-reference/00-architecture/svls-01-architecture.md [hunk 5]
```diff
@@ -6,7 +6,9 @@ Serverless relies heavily on Kubernetes resources. It uses [Deployments](https:/
...
+
+**NOTE:** To enable Istio sidecar proxy injection which is disabled on the Kyma clusters by default, set the `istio-injection=enabled` flag at the Namespace level.
```
Comment (SamehMKhattab) [inline docs/05-technical-reference/00-architecture/svls-01-architecture.md:11]:
```suggestion
**NOTE:** To enable Istio sidecar proxy injection, which is disabled on the Kyma clusters by default, set the `istio-injection=enabled` flag at the Namespace level.
```

**Review by barchw — APPROVED** [review]

**Comment by dariusztutaj** [comment]
Too many issues with raising. PR will be replaced with new one.

**Pull request closed without merging**